packages/dds/merge-tree/src/collections.ts

```diff
@@ -144,7 +144,14 @@ export class List<T> {
         return (this.next === this);
     }
 
+    /**
+     * @deprecated - use unshift
+     */
     public push(data: T): void {
+        this.unshift(data);
+    }
+
+    public unshift(data: T): void {
         const entry = ListMakeEntry(data);
         entry.data = data;
         entry.isHead = false;
```

Comment:
nit: this should be unnecessary, looks like ListMakeEntry already initializes `isHead` to false.

Reply:
yeah. i want to refactor all this code. i find every node being the list super weird to work with, or at least very dangerous.

Reply:
yeah, agree. I was confused how the iterator worked until looking more closely at the structure of the list. It also doesn't really work as-is on an arbitrary node. That doesn't matter based on current usage, though. Would be good to clean up in a followup.